java/java.lsp.server/nbcode/integration/src/org/netbeans/modules/nbcode/integration/LspAssetsDecorationProvider.java

```diff
@@ -85,6 +87,10 @@ public TreeItemData createDecorations(Node n, boolean expanded) {
                 d.addContextValues(CTXVALUE_PREFIX_REFERENCE_NAME + refName);
                 set = true;
             }
+            if (item instanceof ComputeInstanceItem && ((ComputeInstanceItem) item).getPublicIp() != null) {
+                d.addContextValues(CTXVALUE_CAP_PUBLIC_IP);
```

Comment:
an idea: could be the public IP value itself added as the context value on the item (with some prefix to identify) ? Then the LSP client can process the context values, saving some round-trips.